src/MSBuild.UnitTests/XMake_Tests.cs

```diff
@@ -830,13 +830,8 @@ public void GetCommandLineQuotedExe()
             File.Exists(_pathToArbitraryBogusFile).ShouldBeTrue();
 
             bool successfulExit;
-            string pathToMSBuildExe = RunnerUtilities.PathToCurrentlyRunningMsBuildExe;
-            // This @pathToMSBuildExe is used directly with Process, so don't quote it on
-            // Unix
-            if (NativeMethodsShared.IsWindows)
-            {
-                pathToMSBuildExe = "\"" + pathToMSBuildExe + "\"";
-            }
+            string pathToMSBuildExe = Path.GetFullPath(RunnerUtilities.PathToCurrentlyRunningMsBuildExe);
```

Comment:
Unfortunately, I think this invalidates the test, since what we're trying to test here is that our utilities correctly handle both the case where the application part of the command line is quoted and unquoted, that is

```
path\to\msbuild.exe and some arguments
```

and

```
"path\to\msbuild.exe" and some arguments
```

should both work fine.

It looks like the issue is that this test doesn't make sense on .NET Core, because of this code:

https://github.com/dotnet/msbuild/blob/56ab6340935d039999a588621733fa2384dae2a3/src/UnitTests.Shared/RunnerUtilities.cs#L28-L33

On .NET Core we fall into the `else` case there, because instead of running

```
msbuild.exe arguments
```

we run

```
dotnet.exe "path\to\msbuild.dll" arguments
```

As a result, when this test quotes the MSBuild path, it gets double-quoted in the command line

```
dotnet.exe ""path\to\msbuild.dll"" arguments
```

which is basically equivalent to 

```
dotnet.exe path\to\msbuild.dll arguments
```

which fails if the path has a space in it.